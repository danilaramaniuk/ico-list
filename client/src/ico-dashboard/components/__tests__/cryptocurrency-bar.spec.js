import React from 'react';
import { renderWithTheme, getMockedContributions } from '~/__tests__/helpers';
import CryptocurrencyBar from '../cryptocurrency-bar';

jest.mock('react-chartjs-2', () => ({
  Bar: () => <div>Bar</div>,
}));

const props = {
  contributions: getMockedContributions(),
  label: 'label',
  barIndex: 0,
  setBarIndex: jest.fn(),
  color: 'white',
  Icon: () => null,
};

describe('CryptocurrencyBar snapshots', () => {
  it('renders correctly', () => {
    const tree = renderWithTheme(<CryptocurrencyBar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with empty contributions', () => {
    const updatedProps = { ...props, contributions: [] };
    const tree = renderWithTheme(<CryptocurrencyBar {...updatedProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CryptocurrencyBar instance', () => {
  const renderedValue = renderWithTheme(<CryptocurrencyBar {...props} />);
  const { instance } = renderedValue.toTree().rendered;

  it('build data correctly', () => {
    expect(instance.data).toEqual({
      labels: [1, 2, 3],
      datasets: [
        {
          data: [1822819, 2001913, 8288838],
          label: 'label',
          backgroundColor: 'white',
          borderWidth: 1,
        },
      ],
    });
  });

  it('not set bar index when click on empty space', () => {
    instance.onBarClick([]);
    expect(props.setBarIndex).not.toBeCalled();
  });
});
