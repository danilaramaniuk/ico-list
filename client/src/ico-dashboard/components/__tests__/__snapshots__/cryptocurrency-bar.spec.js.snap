// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptocurrencyBar snapshots renders correctly 1`] = `
Array [
  <div
    className="sc-gzVnrw iadGwB"
  >
    <div
      className="sc-EHOje fhxHOS"
    >
       
      <div
        className="sc-bZQynM cELMEx"
      >
        label
      </div>
    </div>
    <div>
      Bar
    </div>
  </div>,
  <div
    className="sc-bdVaJa cxEflh"
  >
    <div
      className="sc-bwzfXH eHRqiU"
    >
      <div
        className="sc-htpNat iNMcUK"
      >
        index
      </div>
      <div
        className="sc-bxivhb kwdKHT"
      >
        1
      </div>
    </div>
    <div
      className="sc-bwzfXH eHRqiU"
    >
      <div
        className="sc-htpNat iNMcUK"
      >
        currency
      </div>
      <div
        className="sc-bxivhb kwdKHT"
      >
        LTC
      </div>
    </div>
    <div
      className="sc-bwzfXH eHRqiU"
    >
      <div
        className="sc-htpNat iNMcUK"
      >
        value
      </div>
      <div
        className="sc-bxivhb kwdKHT"
      >
        1822819
      </div>
    </div>
    <div
      className="sc-bwzfXH eHRqiU"
    >
      <div
        className="sc-htpNat iNMcUK"
      >
        address
      </div>
      <div
        className="sc-htpNat iNMcUK"
      >
        txid
      </div>
    </div>
    <div
      className="sc-ifAKCX gRAkOE"
    >
      <div
        className="sc-bxivhb kwdKHT"
      >
        rnu2xb48synW6Y1AU8ZT85OX0CWBD21N
      </div>
      <div
        className="sc-bxivhb kwdKHT"
      >
        4911922634ee442f6704ed6cf5f860dbe70e78d1c20414d6d6f2eb4bc81e1641
      </div>
    </div>
  </div>,
]
`;

exports[`CryptocurrencyBar snapshots renders correctly with empty contributions 1`] = `null`;
